src/cargo/core/compiler/compilation.rs

```diff
@@ -116,7 +116,7 @@ impl<'cfg> Compilation<'cfg> {
             rustc_process: rustc,
             host: bcx.host_triple().to_string(),
             target: bcx.target_triple().to_string(),
-            target_runner: LazyCell::new(),
```

Comment:
I made this eager which means that in the case of `cfg` collision `cargo run` will *immediately* fail.

Alternatively, if we kept this lazy in that scenario `cargo run` will first build the crate and then emit an error message about the runners (I think).

I don't know if making this eager has any perceivable effects on existing uses of `cargo run`.

Which behavior do we want here?